echopype/tests/visualize/test_plot.py

```diff
@@ -224,21 +226,30 @@ def test_water_level_echodata(water_level, expect_warning):
         ek_waveform_mode=range_kwargs.get('waveform_mode', 'CW'),
         ek_encode_mode=range_kwargs.get('encode_mode', 'power'),
     )
-    single_array = range_in_meter.isel(frequency=0, ping_time=0).values
+
+    # get sorted indices of all frequency_nominal values. This is necessary
+    # because the frequency_nominal values are not always in ascending order.
+    sorted_freq_ind = np.argsort(echodata.vendor.frequency_nominal)
```

Comment:
This function test for the handling and existence of `water_level`, so I think we should have the overarching change to go from indexing the monotonically varying `sorted_freq_ind` to using `channel` itself.